### PR TITLE
Retry when conflict happens in the YCSB workloads

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: scalardb-test/src/main/java/kelpie/scalardb/ycsb/WorkloadA.java [hunk 1]
```diff
@@ -44,32 +52,56 @@ public WorkloadA(Config config) {
...
+          if (useReadModifyWrite) {
+            transaction.get(prepareGet(writeUserId));
+          }
+          transaction.put(preparePut(writeUserId, payloads.get(i)));
         }
-        YcsbCommon.randomFastChars(ThreadLocalRandom.current(), payload);
-        transaction.put(preparePut(userId, new String(payload)));
+        transaction.commit();
+        break;
+      } catch (CrudConflictException | CommitConflictException e) {
```
Comment (brfrn169) [inline scalardb-test/src/main/java/kelpie/scalardb/ycsb/WorkloadA.java:89]: Retry when catching `CrudConflictException` and `CommitConflictException`.

File: scalardb-test/src/main/java/kelpie/scalardb/ycsb/WorkloadA.java [hunk 2]
```diff
@@ -44,32 +52,56 @@ public WorkloadA(Config config) {
...
+        transaction.abort();
+        transactionRetryCount.increment();
```
Comment (brfrn169) [inline scalardb-test/src/main/java/kelpie/scalardb/ycsb/WorkloadA.java:91]: Record the transaction retry count here.

File: scalardb-test/src/main/java/kelpie/scalardb/ycsb/YcsbReporter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            + stats.getLatencyAtPercentile(50.0)
+            + " ms\n"
+            + "Latency at 90 percentile: "
+            + stats.getLatencyAtPercentile(90.0)
+            + " ms\n"
+            + "Latency at 99 percentile: "
+            + stats.getLatencyAtPercentile(99.0)
+            + " ms\n"
+            + "Transaction retry count: "
+            + getPreviousState().getString("transaction-retry-count"));
```
Comment (brfrn169) [inline scalardb-test/src/main/java/kelpie/scalardb/ycsb/YcsbReporter.java:49]: This `YcsbReporter` shows the `transaction-retry-count`.

File: scalardb-test/src/main/java/kelpie/scalardb/ycsb/WorkloadA.java [hunk 4]
```diff
@@ -44,32 +52,56 @@ public WorkloadA(Config config) {
...
     }
   }
 
   @Override
   public void close() throws Exception {
     manager.close();
+    setState(
+        Json.createObjectBuilder()
+            .add("transaction-retry-count", transactionRetryCount.toString())
+            .build());
```
Comment (brfrn169) [inline scalardb-test/src/main/java/kelpie/scalardb/ycsb/WorkloadA.java:105]: Set `transaction-retry-count` to the state to show it in `YcsbReporter`.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**